### PR TITLE
Add api_versions support

### PR DESCRIPTION
**Pull request by kyleknap**
cc @jamesls @JordonPhillips

### CODE REVIEW

File: awscli/clidriver.py [hunk 1]
```diff
@@ -363,8 +363,10 @@ def _get_command_table(self):
 
     def _get_service_model(self):
         if self._service_model is None:
+            api_version = self.session.get_config_variable('api_versions').get(
```
Comment (JordonPhillips) [inline awscli/clidriver.py]: The output of `get_config_variable` is `None` if they don't have versions set, so you'll need to check for that.

**Comment by JordonPhillips** [comment]
Looks good, just one small comment.

File: awscli/clidriver.py (hunk 1 above)
Comment (kyleknap) [inline awscli/clidriver.py]: The default is an empty dictionary: https://github.com/boto/botocore/blob/develop/botocore/session.py#L83. So we can assume that value will always be a dictionary unless they messed up the way in which they specified the value in the config file like: ``api_versions = foo`` which will cause it to be a string.

Comment (JordonPhillips) [inline awscli/clidriver.py]: Ah, I haven't pulled in that change yet. With that in mind, looks good.

**Comment by JordonPhillips** [comment]
:boat:

**Pull request merged**